acceptance/managers/asset_manager.go

```diff
@@ -0,0 +1,313 @@
+		lifecycleDescriptor = a.lifecycleDescriptor
+	case Previous:
+		lifecyclePath = a.previousLifecyclePath
+		lifecycleDescriptor = a.previousLifecycleDescriptor
+	default:
+		a.testObject.Fatalf("lifecycle kind must be default or previous, was %s", kind)
+	}
+
+	return lifecyclePath, lifecycleDescriptor
+}
+
+type assetManagerBuilder struct {
+	testObject  *testing.T
+	inputConfig InputConfigurationManager
+}
+
+func (b assetManagerBuilder) ensureCurrentPack() string {
+	b.testObject.Helper()
+
+	if b.inputConfig.packPath != "" {
+		return b.inputConfig.packPath
+	}
+
+	compileWithVersion := b.inputConfig.compilePackWithVersion
+	if compileWithVersion == "" {
+		compileWithVersion = defaultCompilePackVersion
+	}
+
+	return b.buildPack(compileWithVersion)
+}
+
+func (b assetManagerBuilder) ensurePreviousPack() string {
+	b.testObject.Helper()
+
+	if b.inputConfig.previousPackPath != "" {
+		return b.inputConfig.previousPackPath
+	}
+
+	b.testObject.Logf(
+		"run combinations %+v require %s to be set",
+		b.inputConfig.combinations,
+		style.Symbol(envPreviousPackPath),
+	)
+
+	b.ensureGithubAssetFetcher()
+	version, err := githubAssetFetcher.FetchReleaseVersion("buildpacks", "pack", 0)
+	h.AssertNil(b.testObject, err)
+
+	assetDir, err := githubAssetFetcher.FetchReleaseAsset(
+		"buildpacks",
+		"pack",
+		version,
+		variables.PackBinaryExp,
+		true,
+	)
+	h.AssertNil(b.testObject, err)
+	assetPath := filepath.Join(assetDir, variables.PackBinaryName)
+
+	b.testObject.Logf("using %s for previous pack path", assetPath)
+
+	return assetPath
+}
+
+func (b assetManagerBuilder) ensurePreviousPackFixtures() string {
+	b.testObject.Helper()
+
+	if b.inputConfig.previousPackFixturesPath != "" {
+		return b.inputConfig.previousPackFixturesPath
+	}
+
+	b.testObject.Logf(
+		"run combinations %+v require %s to be set",
+		b.inputConfig.combinations,
+		style.Symbol(envPreviousPackFixturesPath),
+	)
+
+	b.ensureGithubAssetFetcher()
+	version, err := githubAssetFetcher.FetchReleaseVersion("buildpacks", "pack", 0)
+	h.AssertNil(b.testObject, err)
+
+	sourceDir, err := githubAssetFetcher.FetchReleaseSource("buildpacks", "pack", version)
+	h.AssertNil(b.testObject, err)
+
+	fis, err := ioutil.ReadDir(sourceDir)
+	h.AssertNil(b.testObject, err)
+	// GitHub source tarballs have a top-level directory whose name includes the current commit sha.
+	innerDir := fis[0].Name()
+	fixturesDir := filepath.Join(sourceDir, innerDir, "acceptance", "testdata", "pack_fixtures")
+
+	b.testObject.Logf("using %s for previous pack fixtures path", fixturesDir)
+
+	return fixturesDir
+}
+
+func (b assetManagerBuilder) ensureDefaultLifecycle() (string, builder.LifecycleDescriptor) {
+	b.testObject.Helper()
+
+	lifecyclePath := b.inputConfig.lifecyclePath
+
+	if lifecyclePath == "" {
+		b.testObject.Logf(
+			"run combinations %+v require %s to be set",
+			b.inputConfig.combinations,
+			style.Symbol(envLifecyclePath),
+		)
+
+		lifecyclePath = b.downloadLifecycle(0)
```

Comment:
I believe the use of "default" in this PR actually means "current" (by our current definition) - because of this line here. `0` results in us downloading the `n - 0` previous minor version. I question if we want to do away with "default" as it's currently defined (`builder.DefaultLifecycleVersion`). Or maybe since what we are doing is actually downloading the "latest" lifecycle maybe we should call it `latest`.

Reply:
Responding in a [consistent location](https://github.com/buildpacks/pack/pull/708#discussion_r445808594)

Reply:
Although... That said, I do like `latest`